templates/wpcomics/src/imp.rs

```diff
@@ -0,0 +1,620 @@
+		&self,
+		params: &mut Params,
+		url: &str,
+		headers: Option<&[(&str, &str)]>,
+	) -> Result<Request> {
+		// 通常のリクエスト
+		let mut req = Request::new(url, HttpMethod::Get)?;
+		if let Some(cookie) = &params.cookie {
+			req = req.header("Cookie", cookie);
+		}
+		if let Some(user_agent) = params.user_agent {
+			req = req.header("User-Agent", user_agent);
+		}
+		if let Some(extra_headers) = headers {
+			for (key, value) in extra_headers {
+				req = req.header(key, value);
+			}
+		}
+		self.modify_request(params, req)
+	}
+
+	fn category_parser(
+		&self,
+		params: &mut Params,
+		categories: &Vec<String>,
+	) -> (ContentRating, Viewer) {
+		#[allow(clippy::needless_match)]
+		let mut nsfw = params.nsfw;
+		#[allow(clippy::needless_match)]
+		let mut viewer = params.viewer;
+		for category in categories {
+			match category.to_ascii_lowercase().as_str() {
+				"smut" | "mature" | "18+" | "adult" => nsfw = ContentRating::NSFW,
+				"ecchi" | "16+" => {
+					nsfw = match nsfw {
+						ContentRating::NSFW => ContentRating::NSFW,
+						_ => ContentRating::Suggestive,
+					}
+				}
+				"webtoon" | "manhwa" | "manhua" => viewer = Viewer::Webtoon,
+				_ => continue,
+			}
+		}
+
+		(nsfw, viewer)
+	}
+	fn get_manga_list(
+		&self,
+		params: &mut Params,
+		search_url: String,
+		headers: Option<&[(&str, &str)]>,
+	) -> Result<MangaPageResult> {
+		let mut has_next_page = !params.next_page.is_empty();
+
+		let html = self.create_request(params, &search_url, headers)?.html()?;
+
+		let Some(elems) = html.select(params.manga_cell) else {
+			return Ok(MangaPageResult {
+				entries: vec![],
+				has_next_page: false,
+			});
+		};
+		let mut entries: Vec<Manga> = Vec::with_capacity(elems.size());
+		for item_node in elems {
+			let title = item_node
+				.select(params.manga_cell_title)
+				.and_then(|node| node.first())
+				.and_then(|n| n.text());
+			let url = item_node
+				.select(params.manga_cell_url)
+				.and_then(|node| node.first())
+				.and_then(|n| n.attr("abs:href"))
+				.unwrap_or("".to_string());
+
+			let cover = if !params.manga_cell_image.is_empty() {
+				item_node
+					.select(params.manga_cell_image)
+					.and_then(|v| v.first())
+					.and_then(|n| n.attr(params.manga_cell_image_attr))
+			} else {
+				None
+			};
+			entries.push(Manga {
+				key: (params.manga_parse_id)(url).to_string(),
+				cover,
+				title: (params.manga_details_title_transformer)(title.unwrap_or("".to_string()))
+					.to_string(),
+				..Default::default()
+			});
+		}
+		if !params.next_page.is_empty() {
+			has_next_page = html
+				.select(params.next_page)
+				.map(|v| v.size() > 0)
+				.unwrap_or(false);
+		}
+		Ok(MangaPageResult {
+			entries,
+			has_next_page,
+		})
+	}
+
+	fn parse_manga_element(&self, params: &mut Params, url: String) -> Result<Manga> {
+		self.cache_manga_page(params, url.as_str())?;
+
+		let html = self.get_cache_manga_value(params).unwrap();
+		let details = Html::parse_with_url(html, url.clone())?;
```

Comment:
the url shouldn't need to be cloned
```suggestion
		let details = Html::parse_with_url(html, &url)?;
```